snuba/migrations/snuba_migrations/metrics/0020_polymorphic_buckets_table.py

```diff
@@ -0,0 +1,74 @@
+from typing import Sequence
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.utils.schemas import Array, Column, DateTime, Float, Nested, String, UInt
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Create a polymorphic bucket table for metrics_sets,
+    metrics_distributions, and metrics_counters
+    """
+
+    blocking = False
+    local_table_name = "metrics_buckets_consolidated"
+    dist_table_name = "metrics_buckets_consolidated_dist"
+
+    column_list: Sequence[Column[Modifiers]] = [
+        Column("org_id", UInt(64)),
+        Column("project_id", UInt(64)),
+        Column("metric_id", UInt(64)),
+        Column("timestamp", DateTime()),
+        Column("tags", Nested([Column("key", UInt(64)), Column("value", UInt(64))]),),
+        Column("metric_type", String(Modifiers(low_cardinality=True))),
+        Column("set_values", Array(UInt(64))),
+        Column("count_value", Float(64)),
```

Comment:
Functionally:
I would assume your materialized views will always filter by metric_type. So whether this is nullable or not does not matter. Could you confirm ? 
Storage:
Could you check whether a column full of 0.0 float64 is bigger than a nullable column full of NULL ?
You can do that locally by creating a table with a ton of values and check the `bytes_on_disk` field in the `system.columns` table.

Reply:
a column filled with 0.0 (or any float) is definitely much smaller than a column filled with NULLs, see some testing I did with https://gist.github.com/onewland/20799a8feee3317a1a6f6061f28297a9

Reply:
This result may have actually been due to bad partitioning, I'll update in a minute